CRM/Contact/BAO/Query.php

```diff
@@ -588,14 +592,18 @@ public function buildParamsLookup() {
         $this->_paramLookup[$value[0]] = array();
       }
       if ($value[0] !== 'group') {
-        // Just trying to unravel how group interacts here! This whole function is wieid.
+        // Just trying to unravel how group interacts here! This whole function is weird.
         $this->_paramLookup[$value[0]][] = $value;
       }
     }
   }
 
   /**
    * Some composite fields do not appear in the fields array hack to make them part of the query.
+   *
+   * @param $apiEntity
+   *   The api entity being called.
+   *   This sort-of duplicates $mode in a confusing way. Probably not by design.
```

Comment:
This looks nicer.

A suggestion on docblock style going-forward (not meant as a complaint about this): `@param $apiEntity` looks like a typical case where I'd try to include an example as part of the doc, as in:

```
 * @param string $apiEntity
 *   The api entity being called. Ex: 'Contact' or 'ContributionPage'.
```

As readers who are tuned-in to the nuances of Civi's data modelling (SQL/DAO/BAO/API), you and I probably interpret `apiEntity` as "an entity name in the camel-case notation of APIv3" (`ContributionPage`), but less experienced contributors might interpret "entity" more ambiguously. (`ContributionPage`, `contribution_page`, `api/v3/ContributionPage.php`, `civicrm_api3_contribution_page_*`, `CRM_Contribute_DAO_ContributionPage`, and `civicrm_contribution_page` are all ways to identify the same entity... and all those ways would actually show up if you were tracing an API call.)

An example value is a really simple/clear.